src/main/java/com/firebolt/jdbc/connection/UrlUtil.java

```diff
@@ -0,0 +1,61 @@
+package com.firebolt.jdbc.connection;
+
+import com.firebolt.jdbc.connection.settings.FireboltSessionProperty;
+import lombok.CustomLog;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Optional;
+import java.util.Properties;
+
+@CustomLog
+@UtilityClass
+public class UrlUtil {
+
+    public static final String JDBC_PREFIX = "jdbc:firebolt:";
+
+    public static Properties extractProperties(String jdbcUrl) {
+        return parseUriQueryPart(jdbcUrl);
+    }
+
+
+    private static Properties parseUriQueryPart(String jdbcConnectionString) {
```

Comment:
Is there no standard url parser we can use? Why do we have a custom parser here?

Reply:
This code was written on 22.03.2023; I have not changed code I do not have to change. Also unfortunately JDK does not provide function that parses URL parameters and return query parameters as properites, map or something like this.